source/tutorials/batch-inference/image-batch-inference.rst

```diff
@@ -0,0 +1,169 @@
+
+.. admonition:: Prerequirements
+    :class: attention
+
+    For this tutorial you will need:
+
+    * Python 3
+    * Valohai command-line client (Run ``pip install --upgrade valohai-cli``)
+
+    We're also going to need two files:
+
+    * an MNIST model trained with TensorFlow 2.4.1
+    * some images to classify, packaged in a single zip file
+
+    To make it easy for you they are available here, no need to download them:
+
+    * Model: `<s3://valohai-public-files/tutorials/batch-inference/image-batch-inference/model.h5>`_
+    * Images: `<s3://valohai-public-files/tutorials/batch-inference/image-batch-inference/images.zip>`_
+
+    If you want to, you can train the required model by following our `Valohai 101 tutorial </tutorials/quickstart/>`_.
+
+Running on Valohai
+------------------
+
+To easily run our batch inference on Valohai, we will use it to run our code from the very beginning.
+
+If you don't already have a Valohai account, go to `<https://app.valohai.com/>`_ to create one for yourself.
+
+Create a new folder for our project, then run the following commands in the project folder:
+
+.. code-block:: bash
+
+    vh login
+    # fill in your username
+    # and your password
+
+    vh init
+    # Answer the wizard questions like this:
+    # "First, let's..." -> y
+    # "Looks like..." -> python batch_inference.py, then y to confirm
+    # "Choose a number or..." -> tensorflow/tensorflow:2.4.1, then y to confirm
+    # "Write this to..." -> y
+    # "Do you want to link..." -> C, then give a name for your project, then select your user
+
+Edit the generated ``valohai.yaml`` so that it looks like this:
+
+.. code-block:: yaml
+
+    ---
+
+    - step:
+        name: Batch Inference
+        image: tensorflow/tensorflow:2.4.1
+        command:
+        - pip install pillow valohai-utils
+        - python batch_inference.py
+        inputs:
+        - name: model
+          default: s3://valohai-public-files/tutorials/batch-inference/image-batch-inference/model.h5
+        - name: images
+          default: s3://valohai-public-files/tutorials/batch-inference/image-batch-inference/images.zip
+
+What we are doing here is defining a single step for our machine learning pipeline, which is the Batch Inference step. We will run on top of the official ``tensorflow/tensorflow:2.4.1`` Docker image, first install the ``pillow`` and ``valohai-utils`` Python libraries and then run our batch inference code.
+
+Let's test that everything is set up correctly by running on Valohai:
+
+.. code-block:: bash
+
+    vh exec run --adhoc "Batch Inference"
+
+If everything went as planned, we should see our Valohai execution end after finding out that ``batch_inference.py`` is missing:
+
+.. image:: batch-inference-tutorial-1.png
+   :alt: Error, but success!
+
+Unpacking the Images
+--------------------
+
+...actually Valohai unpacks the images automatically, so we can just fetch their location with ``vh.inputs('images').paths()``!
```

Comment:
This should be explained better. It's not Valohai unpacking images, it's the `valohai-utils` library, when `process_archives` is on...